platform/tests/UNITTESTS/doubles/CMakeLists.txt

```diff
@@ -17,14 +26,17 @@ target_sources(mbed-stubs-platform
         randLIB_stub.c
         randLIB_stub.cpp
 )
-
+target_include_directories(mbed-stubs-platform
+    PUBLIC
+        .
+)
 target_link_options(mbed-stubs-platform
     PRIVATE
         --coverage
 )
```

Comment:
Given that `--coverage` is globally set in UNITTESTS/stubs/CMakeLists.txt, can we remove it from here?

Reply:
I'd assume we can as we set it on `CMAKE_CXX_FLAGS` directly in UNITTESTS/CMakeLists.txt. I'll look into removing it as part of this PR.

Reply:
This extra flag exists only in platform and hal (apart from UNITTESTS/CMakeLists.txt), so I assume they are leftovers or something...